util/src/main/java/tc/oc/pgm/util/menu/items/InventoryItemBuilder.java

```diff
@@ -0,0 +1,115 @@
+package tc.oc.pgm.util.menu.items;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.pgm.util.menu.InventoryMenu;
+import tc.oc.pgm.util.menu.InventoryMenuManager;
+
+public class InventoryItemBuilder {
+  // TODO: Add back button
+  private final InventoryMenuManager manager;
+  private final BiFunction<InventoryMenu, Player, ItemStack> createItem;
```

Comment:
Anytime you have a builder with a complex function or manager passed around.. it's probably better to just make a static class in the settings module: `SettingItem`. I'd say you don't need the builder.

Reply:
What do you mean by that? Should I create some sort of "creator" interface which serves the same purpose as the builder?

Reply:
My suggestion is instead of having `InventoryItemBuilder` and `InventoryMenuBuilder`, just define your own `SettingsItem` and `SettingsMenu`. Likely that will simplify constructors, also each menu has its own item rendering and slot distribution strategy.